### PR TITLE
[ADF-3362][ADF-3361] Search on document picker with custom site list - fix

### PR DESCRIPTION
**Pull request by suzanadirla**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [x] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [x] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
Please see https://issues.alfresco.com/jira/browse/ADF-3362 and https://issues.alfresco.com/jira/browse/ADF-3361

**What is the new behaviour?**
Fixes issues https://issues.alfresco.com/jira/browse/ADF-3362 and https://issues.alfresco.com/jira/browse/ADF-3361

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by eromano** [comment]
Two things:
1. can you reactor this method to have only one return?
2. Where is used this method without passing the nodeId?

**Comment by suzanadirla** [comment]
Ok, @eromano, I’m working on a refactoring at the moment.

File: lib/content-services/content-node-selector/content-node-selector-panel.component.ts [hunk 1]
```diff
@@ -255,8 +257,8 @@ export class ContentNodeSelectorPanelComponent implements OnInit, PaginatedCompo
     private querySearch(): void {
         this.loadingSearchResults = true;
 
-        if (this.dropdownSiteList) {
-            this.documentList.getCorrespondingNodeIds(this.siteId)
+        if (this.customResourcesService.hasCorrespondingNodeIds(this.siteId)) {
+            this.customResourcesService.getCorrespondingNodeIds(this.siteId, { maxItems: 999 })
```
Comment (eromano) [inline lib/content-services/content-node-selector/content-node-selector-panel.component.ts]: magic number 999 we should never add this kind of fix number in the code

**Comment by eromano** [comment]
without any test is not clear to me which problem we are trying to address

**Comment by suzanadirla** [comment]
I tried to remove the getCorrespondingNodeIds method and use the loadFolderByNodeId method instead... but the problem is that the IDs for custom sources are in different properties: for example for `sharedLinks` it is the `nodeId`, for the sites is the `guid`, for the favorites it is the `targetGuid` :neutral_face: so I need to handle these cases and would end up with another method for this.
In this case, I don’t think it is worth the effort to change the getCorrespondingNodeIds so that it uses the loadFolderByNodeId inside...

**Pull request merged**